codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/SymbolVisitor.kt

```diff
@@ -137,10 +144,14 @@ class SymbolVisitor(
     }
 
     private fun handleOptionality(symbol: Symbol, member: MemberShape, container: Shape): Symbol {
-        // If a field has the httpLabel trait and we are generating
+        // If a field has the httpLabel or hostLabel trait and we are generating
         // an Input shape, then the field is _not optional_.
         val httpLabeledInput =
-            container.hasTrait(SyntheticInputTrait::class.java) && member.hasTrait(HttpLabelTrait::class.java)
+            (container.hasTrait(SyntheticInputTrait::class.java) || container.hasTrait(InputBodyTrait::class.java)) && (
+                member.hasTrait(HttpLabelTrait::class.java) || member.hasTrait(
+                    HostLabelTrait::class.java
+                )
```

Comment:
We defined many of these already FYI (has/expect/get/etc), feel free to steal them (`model/ShapeExt.kt`)

```kt
/**
 * Kotlin sugar for hasTrait() check. e.g. shape.hasTrait<EnumTrait>() instead of shape.hasTrait(EnumTrait::class.java)
 */
inline fun <reified T : Trait> Shape.hasTrait(): Boolean = hasTrait(T::class.java)
```

Separate PR of course

Reply:
Refactored in #385